Mathlib/GroupTheory/Subgroup/Basic.lean

```diff
@@ -3498,6 +3508,19 @@ instance (priority := 100) Subgroup.normal_subgroupOf {H N : Subgroup G} [N.Norm
 #align subgroup.normal_subgroup_of Subgroup.normal_subgroupOf
 #align add_subgroup.normal_add_subgroup_of AddSubgroup.normal_addSubgroupOf
 
+theorem Subgroup.map_normalClosure (s : Set G) (f : G →* N) (hf : Surjective f) :
+    (normalClosure s).map f = normalClosure (f '' s) := by
+  have : Normal (map f (normalClosure s)) := Normal.map inferInstance f hf
+  apply le_antisymm
+  · simp [map_le_iff_le_comap, normalClosure_le_normal, coe_comap,
+      ← Set.image_subset_iff, subset_normalClosure]
+  · exact normalClosure_le_normal (Set.image_subset f subset_normalClosure)
+
+theorem Subgroup.comap_normalClosure (s : Set N) (f : G ≃* N) :
+    normalClosure (f ⁻¹' s) = (normalClosure s).comap f := by
```

Comment:
Maybe this is best saved for another PR, but you can weaken this to `Surjective f` under the assumption that `s` is `Nonempty`:
```
theorem Subgroup.ker_le_closure_preimage_of_nonempty (s : Set N)
    (f : G →* N) (hs : (f ⁻¹' s).Nonempty) : f.ker ≤ closure (f ⁻¹' s) := by
  obtain ⟨g, hg⟩ := hs
  refine' le_trans (fun k h ↦ _) (closure_mono (Set.preimage_mono (Set.singleton_subset_iff.2 hg)))
  rw [← mul_inv_cancel_right k g]
  refine' mul_mem (subset_closure _) (inv_mem (subset_closure _)) <;> simp [map_mul, f.mem_ker.1 h]

theorem Subgroup.comap_normalClosure_of_nonempty (s : Set N) (hs : s.Nonempty)
    (f : G →* N) (hf : Surjective f) :
    normalClosure (f ⁻¹' s) = (normalClosure s).comap f := by
  refine' map_injective_of_ker_le f
    ((ker_le_closure_preimage_of_nonempty s f (hs.preimage hf)).trans closure_le_normalClosure)
    (ker_le_comap f (normalClosure s)) _
  rw [map_normalClosure (f ⁻¹' s) f hf, map_comap_eq_self_of_surjective hf,
    Set.image_preimage_eq s hf]
```

Reply:
@tb65536 I had looked for a proof to weaken the equivalence assumption so I am glad to see this!  

To further the sake of symmetry among the lemmas it may be possible to remove the `NonEmpty` from the assumptions and then do a proof `by_cases H : s.NonEmpty` *if* when `s` is empty you can derive a contradiction.  It is getting in the wee hours of the morning here so I may take a stab at it in the morning but feel free to add this if you think this is a decent idea (if not, I would like to hear back on the reasons so I can learn why) and go ahead and push a commit.  

Thanks for the feedback!

Reply:
@eric-wieser what should be done here?  I want to get this PR across the line as I have a couple of branches blocked on it.

Reply:
Given this is not a strict generalization, these should be in addition to the existing lemmas not instead of them. I don't care whether they are in this PR or a follow-up.

Reply:
I say we just keep them in a separate PR then.